### DISCUSSION TITLE
Is it possible to have a Socket.io connection between two backend files, with no Client involved?

### DISCUSSION THREAD

**Discussion by respectabiggle** [socketio/socket.io#4941] [Q&A] [answered]
Thank you!

Starting code here:  https://github.com/respectabiggle/socketIoServerToServer

**Comment by darrachequesne** [comment, marked as answer]
Yes! One server can act as a client, and the other one as an actual server.

You can also have the two servers act as servers, and then use `io.serverSideEmit()` to exchange data between them.

Reference: https://socket.io/docs/v4/server-instance/#serversideemit

**Reply by respectabiggle** [reply]
I so much appreciate you being present and answering my above question.  You also really helped me [last year](https://github.com/socketio/socket.io/discussions/4490#discussioncomment-3827837).

1 more question, please:  

I cannot find the emitter initialization for the Cluster Adapter version of the adapter:   https://socket.io/docs/v4/cluster-adapter/

(I see emitter initializations at the other adapter methods, [Redis](https://github.com/socketio/socket.io/discussions/4490#discussioncomment-3827837) etc).

Is it missing from the docs?  Or am I just lacking knowledge?

Should I be using https://github.com/socketio/emitter?

**Reply by darrachequesne** [reply]
No, there is currently no emitter implementation, but this shouldn't be too hard to implement. Could you please explain your use case?

Possibly related: https://github.com/socketio/socket.io-cluster-adapter/issues/10

**Reply by respectabiggle** [reply]
Thanks very much @darrachequesne 

Use case:  

I'm trying to pass data between two files, neither of which is client.js.  It is all happening strictly on the back-end.  There is no web interface involved.  I would like to pass that data via a websocket, rather than through API calls.


Fuller description:

My app listens to an external websocket via ws.

The incoming streaming data is used in several files in the app.

Rather than duplicating the ws connection to the external websocket in each of those files, 
I would like to emit that stream via a Socket.io channel from one file 

> externalWebsocket.js

and be able to "listen in" on that channel from anywhere else in my app.

> useCase1.js
> useCase2.js

> note:  even though the example github repository I liked in the OP is named "server to server", my app doesn't actually involve more than one server.  I just named it such to distinguish it from the typical "server to client" setup.  Is therefore the "multi-server-setup"/clusters/adapters linked from https://socket.io/docs/v4/server-instance/#serversideemit still the way to go?

Thanks again
If I am unclear please let me know

**Comment by respectabiggle** [comment]
Man!  Thank you so much!
